packages/core/src/middleware/alchemy-paymaster.ts

```diff
@@ -26,6 +46,66 @@ export interface AlchemyPaymasterConfig {
   provider: PublicErc4337Client;
 }
 
+/**
+ * This uses the alchemy RPC method: `alchemy_requestGasAndPaymasterAndData` to get all of the gas estimates + paymaster data
+ * in one RPC call. It will no-op the gas estimator and fee data getter middleware and set a custom middleware that makes the RPC call
+ *
+ * @param provider - the smart account provider to override to use the alchemy paymaster
+ * @param config - the alchemy paymaster configuration
+ * @returns the provider augmented to use the alchemy paymaster
+ */
+export const useAlchemyGasManager = <
```

Comment:
we might want to name this something different as a frontend dev would def confuse this for a react hook.

Reply:
maybe genAlchemyGasManager or createAlchemyGasManager or createAlchemyGasManagerMiddleware

Reply:
ah yea good call, I'll change it to `withAlchemyGasManager` since it's meant to be like `provider with AlchemyGasManager`